### PR TITLE
test(driver-adapters-wasm): vercel edge functions

### PR DESCRIPTION
**Pull request by millsp**
This PR tests Vercel edge functions via the nextjs app router and page router. We are only adding tests for neon, as we are already testing all driver adapters via Vercel edge middleware, so this more of a "check" to validate that it works as expected in other variants of edge deployments as well.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -1183,7 +1183,7 @@ jobs:
 
   driver-adapters:
     needs: [detect_jobs_to_run]
-    if: contains(fromJSON(needs.detect_jobs_to_run.outputs.jobs), 'driver-adapters')
+    if: false && contains(fromJSON(needs.detect_jobs_to_run.outputs.jobs), 'driver-adapters')
```
Comment (janpio) [inline .github/workflows/test.yaml]: We should figure out a way how one can add a new project to the test.yaml without needing to do this here.

File: driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# vercel-edge-functions
+
+Deploys a Prisma Client using the Neon Adapter on Vercel Edge Function (via Pages).
```
Comment (janpio) [inline driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/README.md]: What does "(via Pages)" actually refer to here? I only know "Cloudflare Pages".

Comment (millsp) [reply driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/README.md]: Next.js has two modes: either via the `pages` or the `app` folder. We should test both.

File: driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/next.config.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/**
+ * @type {import('next').NextConfig}
+ */
+const nextConfig = {
+  webpack(config) {
+    config.experiments ??= {}
+
+    config.experiments.asyncWebAssembly = true
```
Comment (janpio) [inline driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/next.config.js:8]: Seeing this again, this is definitily a smell and potential problem.
Let's open an EA issue.

File: .github/workflows/test.yaml (hunk 1 above)
Comment (millsp) [reply .github/workflows/test.yaml]: Yep, it's a tiny bit annoying and potentially dangerous

File: driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/README.md (hunk 2 above)
Comment (janpio) [reply driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/README.md]: Ah, "Pages Router" then for me :D

File: driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/next.config.js (hunk 3 above)
Comment (millsp) [reply driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/next.config.js:8]: Since this is out of our control, would this be about documenting it?

File: .github/workflows/test.yaml (hunk 1 above)
Comment (janpio) [reply .github/workflows/test.yaml]: Tech debt issue?

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -1333,6 +1335,10 @@ jobs:
       DRIVER_ADAPTERS_NEON_CF_BASIC_DATABASE_URL: ${{ secrets.DRIVER_ADAPTERS_NEON_CF_BASIC_DATABASE_URL }}
       DRIVER_ADAPTERS_NEON_VERCEL_NEXTJS_EDGEMW_DATABASE_URL: ${{ secrets.DRIVER_ADAPTERS_NEON_VERCEL_NEXTJS_EDGEMW_DATABASE_URL }}
       DRIVER_ADAPTERS_NEON_VERCEL_NEXTJS_EDGEMW_PROJECT_ID: ${{ vars.DRIVER_ADAPTERS_NEON_VERCEL_NEXTJS_EDGEMW_PROJECT_ID }}
+      DRIVER_ADAPTERS_NEON_VERCEL_NEXTJS_PAGES_EDGEFN_DATABASE_URL: ${{ secrets.DRIVER_ADAPTERS_NEON_VERCEL_NEXTJS_PAGES_EDGEFN_DATABASE_URL }}
```
Comment (janpio) [inline .github/workflows/test.yaml:1338]: OT: We should figure out a way how we can use the standard database that we use for all local tests for these external tests as well. Quite a few secrets would become free with that.

File: driver-adapters-wasm/neon-vercel-nextjs-app-edgefn/app/api/index/route.js [hunk 5]
```diff
@@ -0,0 +1,170 @@
+// @ts-check
+import { Prisma, PrismaClient } from '@prisma/client/edge'
+import { Pool } from '@neondatabase/serverless'
+import { PrismaNeon } from '@prisma/adapter-neon'
+
+export const runtime = 'edge';
```
Comment (janpio) [inline driver-adapters-wasm/neon-vercel-nextjs-app-edgefn/app/api/index/route.js:6]: This is what makes the serverless function to run as a edge function?

File: driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/README.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# neon-vercel-nextjs-pages-edgefn
+
+Deploys a Prisma Client using the Neon Adapter on Vercel Edge Function (via app router).
```
Comment (janpio) [inline driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/README.md]:
```suggestion
Deploys a Prisma Client using the Neon Adapter on Vercel Edge Function (via pages router).
```

File: driver-adapters-wasm/neon-vercel-nextjs-app-edgefn/app/api/index/route.js (hunk 5 above)
Comment (millsp) [reply driver-adapters-wasm/neon-vercel-nextjs-app-edgefn/app/api/index/route.js:6]: Yes

File: driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/next.config.js (hunk 3 above)
Comment (janpio) [reply driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/next.config.js:8]: Yes - deeply understanding, anticipating how our users will discover or hit it, and then document to avoid it leading to pain and failure.

File: driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/pages/api/index.js [hunk 7]
```diff
@@ -0,0 +1,172 @@
+// @ts-check
+import { Prisma, PrismaClient } from '@prisma/client/edge'
+import { Pool } from '@neondatabase/serverless'
+import { PrismaNeon } from '@prisma/adapter-neon'
+
+export const config = {
+  runtime: 'edge',
+}
```
Comment (janpio) [inline driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/pages/api/index.js:8]: Same question: Without this, this runs as a normal serverless function?

Comment (millsp) [reply driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/pages/api/index.js:8]: Yes

File: .github/workflows/test.yaml (hunk 1 above)
Comment (millsp) [reply .github/workflows/test.yaml]: https://github.com/prisma/team-orm/issues/676

File: packagers/yarn3-workspaces-pnp/yarn.lock
Comment (janpio) [inline packagers/yarn3-workspaces-pnp/yarn.lock:1]: Unrelated

File: driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/next.config.js (hunk 3 above)
Comment (millsp) [reply driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/next.config.js:8]: https://github.com/prisma/team-orm/issues/677

Comment (jkomyno) [reply driver-adapters-wasm/neon-vercel-nextjs-pages-edgefn/next.config.js:8]: Could it be helpful to share with users that this is just Webpack's equivalent of Node.js' `--experimental-wasm-modules`, or Vite's `vite-plugin-wasm`? For self-preservation, I think it's important to get the point across to users: these extra configurations steps required to run Prisma on Edge Functions aren't due to bad DX in Prisma itself

**Review by jkomyno — APPROVED** [review]
